python/brayns/client.py

```diff
@@ -0,0 +1,226 @@
+#
+# This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+#
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+# All rights reserved. Do not distribute without further notice.
+
+"""Client that connects to a remote running Brayns instance which provides the supported API."""
+
+import base64
+import io
+
+from PIL import Image
+from .api_generator import build_api
+from .rpcclient import RpcClient
+from .utils import in_notebook
+
+from .utils import HTTP_METHOD_GET, HTTP_STATUS_OK
+from . import utils
+
+
+class Client(RpcClient):
+    """Client that connects to a remote running Brayns instance which provides the supported API."""
+
+    def __init__(self, url):
+        """
+        Create a new client instance by connecting to the given URL.
+
+        :param str url: a string 'hostname:port' to connect to a running brayns instance
+        """
+        super(Client, self).__init__(url)
+        self._check_version()
+        build_api(self, self.url())
+
+        if in_notebook():
+            self._add_widgets()  # pragma: no cover
+
+    def __str__(self):
+        """Return a pretty-print on the currently connected Brayns instance."""
+        # pylint: disable=E1101
+        version = 'unknown'
+        if self.version:
+            version = '.'.join(str(x) for x in [self.version.major, self.version.minor,
+                                                self.version.patch, self.version.revision])
+        return "Brayns version {0} running on {1}".format(version, self.url())
+
+    # pylint: disable=W0613,W0622,E1101
+    def image(self, size, format='jpg', quality=None, samples_per_pixel=None):
+        """
+        Request a snapshot from Brayns and return a PIL image.
+
+        :param tuple size: (width,height) of the resulting image
+        :param str format: image type as recognized by FreeImage
+        :param int quality: compression quality between 1 (worst) and 100 (best)
+        :param int samples_per_pixel: samples per pixel to increase render quality
+        :return: the PIL image of the current rendering, None on error obtaining the image
+        :rtype: :py:class:`~PIL.Image.Image`
+        """
+        args = locals()
+        del args['self']
+        result = self.snapshot(response_timeout=None, **{k: v for k, v in args.items() if v})
```

Comment:
I imagine `.snapshot()` is one of the dynamically generated fns. If that's the case, could this `.image()` fn accept all that the snapshot would (renderer, etc.)?